src/server/api/routers/user.ts

```diff
@@ -0,0 +1,220 @@
+import { z } from "zod";
+import {
+  createTRPCRouter,
+  protectedProcedure,
+  workspaceProcedure,
+} from "~/server/api/trpc";
+import { workspaceAccessMiddleware } from "./workspace";
+import { selectWorkspaceUserSchema } from "~/types/workspace_user";
+import {
+  userInviteTable,
+  userTable,
+  workspaceTable,
+  workspaceUserTable,
+} from "~/server/db/schema";
+import { and, eq } from "drizzle-orm";
+import {
+  insertUserInviteSchema,
+  selectUserInviteSchema,
+  selectUserSchema,
+} from "~/types/user";
+import { selectWorkspaceSchema } from "~/types/workspace";
+import { render } from "@react-email/render";
+import { WorkspaceUserInvite } from "~/emails/workspace-user-invite";
+import { sendEmailWithSES } from "~/lib/email";
+import { TRPCError } from "@trpc/server";
+import { env } from "~/env";
+import _ from "lodash";
+
+export const userRouter = createTRPCRouter({
+  getUsersInWorkspace: workspaceProcedure
+    .input(z.object({ workspaceId: z.string() }))
+    .use(workspaceAccessMiddleware)
+    .output(
+      z.array(
+        z.object({
+          user: selectUserSchema,
+          workspace: selectWorkspaceSchema,
+          workspace_user: selectWorkspaceUserSchema,
+        }),
+      ),
+    )
+    .query(({ ctx, input }) => {
+      const result = ctx.db
+        .select()
+        .from(workspaceUserTable)
+        .where(eq(workspaceUserTable.workspaceId, input.workspaceId))
+        .innerJoin(userTable, eq(userTable.id, workspaceUserTable.userId))
+        .innerJoin(
+          workspaceTable,
+          eq(workspaceTable.id, workspaceUserTable.workspaceId),
+        );
+      return result;
```

Comment:
can't you just do a groupBy here so you don't have repeating workspace objects in each array entry?